cmd/system-probe/modules/compile.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package modules
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	rc "github.com/DataDog/datadog-agent/pkg/runtimecompiler"
+	compilerconfig "github.com/DataDog/datadog-agent/pkg/runtimecompiler/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+)
+
+const (
+	statsdPoolSize = 64
+)
+
+// Compiler Factory
+var Compiler = module.Factory{
```

Comment:
What do you think about each module having a `CompileFn` separate from the `Fn`? That way on the `compile` command, the loader just runs each module's `CompileFn`. I think this might also allow a better separation of concerns with the compiler. I don't like how it currently has to know about all the modules, their configs, and their compilation needs in the same place.